### PR TITLE
[Review Only] Proposing encryption block structure per object

### PR DESCRIPTION
**Pull request by zed-rishabh**
These changes are proposing the structure of the encryption block for each object. The current structure is restricted to only the credentials of the objects. If we want to add the encryption for user data in the app config then we need to add a new field that will be available for all the objects (not required).

The plan is the different agents will request for decryption to the TPM manager and tpmmgr will return the decrypted data. And, each agent will take the decrypted data and populate it in its own required structure.

### CODE REVIEW

**Comment by zed-rishabh** [comment]
@cshari-zededa @saurabh-zededa Please take a look.

**Comment by rvs** [comment]
Honestly @zed-rishabh -- I'd like to avoid the creep of credentials as much as possible -- what's the issue of keeping it all in a common block and just everything offset into that?

**Comment by zed-rishabh** [comment]
> I don't think we need separate structures, because it means that the code will be separate to handle that struct. The strings in credentialblock can be set to empty if not used for some particular secure blob. So either using "identity" for the user-data, or defining a new "protectedData" string in CredentialBlock.
> 
> In either case comments can explain that different subsets of CredentialBlock are used for the different cases.

Okay, I will update the PR.

File: api/proto/config/acipherinfo.proto [hunk 1]
```diff
@@ -64,10 +64,13 @@ message CipherBlock {
 
 
 // This message will be filled with the
-// credential details and encrypted across
-//  wire for data in transit, by the controller
+// details to be encrypted and shared across
+// wire for data in transit, by the controller
 // for encryption
```
Comment (eriknordmark) [inline api/proto/config/acipherinfo.proto:69]: I think you need to comment that different subsets of these fields are filled in for different usage of this block.
Doesn't this mean that you'll end up with different code to handle the decrypt? Or the different use cases pick up different fields.

Comment (cshari-zededa) [reply api/proto/config/acipherinfo.proto:69]: Hi @eriknordmark,  the decrypt logic will be common, and give back the structure, and individual agents will pick up different fields post decryption.

**Comment by zed-rishabh** [comment]
> LGTM
> 
> To you want me to pull this, or do you want to add the derived and vendored files first?

Do not merge this one. I will raise another PR with all the changes in two commits.

**Comment by zed-rishabh** [comment]
Please take a look https://github.com/lf-edge/eve/pull/845.

**Pull request closed without merging**